src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp

```diff
@@ -46,36 +46,41 @@ bool emberAfAdministratorCommissioningClusterOpenCommissioningWindowCallback(
     auto & salt                 = commandData.salt;
     auto & passcodeID           = commandData.passcodeID;
 
-    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    uint8_t status = 0;
```

Comment:
We should use the right type here: `StatusCode`, given that we are `using namespace chip::app::Clusters::AdministratorCommissioning`.

And maybe what we should do is make this an `Optional<StatusCode>`, have the error paths do `status.Emplace(whatever)` and then down below check for `status.HasValue()`, so we are not making assumptions about 0 being a status value that can't happen....

Similar for other functions.

Reply:
Originally I tried using StatusCode as the type, but it's guarded under `#ifdef CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
`. Looks like `CHIP_USE_ENUM_CLASS_FOR_IM_ENUM` is not set. Is that not expected?

Reply:
```
#ifdef CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
// Enum for StatusCode
enum class StatusCode : uint8_t
{
    kSuccess      = 0x00,
    kBusy         = 0x01,
    kGeneralError = 0x02,
};
#else // CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
using StatusCode                           = EmberAfStatusCode;
#endif
```

Reply:
So the point is, StatusCode is defined no matter what.  The question is what the values look like and whether it's an enum class or not.

If you use `StatusCode::EMBER_ZCL_STATUS_CODE_BUSY` for values and `StatusCode` to declare instances now, that reduces the amount of change we need later when we make `StatusCode` the cluster-objects enum class (to just renaming the values, not having to change the types)...